internal/server/data/grant.go

```diff
@@ -223,9 +228,11 @@ type GrantsMaxUpdateIndexOptions struct {
 func GrantsMaxUpdateIndex(tx ReadTxn, opts GrantsMaxUpdateIndexOptions) (int64, error) {
 	query := querybuilder.New("SELECT max(update_index) FROM grants")
 	query.B("WHERE organization_id = ?", tx.OrganizationID())
-
-	if opts.ByDestination != "" {
-		grantsByDestination(query, opts.ByDestination)
+	if opts.ByDestinationName != "" {
+		query.B("AND destination_name = ?", opts.ByDestinationName)
+		if opts.ByDestinationResource != "" {
```

Comment:
We should error if `ByDestinationResource` is specified but `ByDestinationName` isn't to catch developer errors (same for the delete case).